src/test/java/org/apache/platypus/server/YelpSuggestTest.java

```diff
@@ -0,0 +1,273 @@
+                .setIndexName(INDEX_NAME)
+                .setMode(Mode.STANDALONE)
+                .setPrimaryGen(0)
+                .setRestore(false)
+                .build();
+        startIndex(standaloneServerClient, startIndexRequest);
+        // index docs
+        long t1 = System.nanoTime();
+        Stream.Builder<AddDocumentRequest> builder = Stream.builder();
+
+        final ExecutorService indexService = YelpReviewsTest.createExecutorService(
+                (Runtime.getRuntime().availableProcessors()) / 4,
+                "LuceneIndexing");
+
+        Path suggestionsPath = Paths.get(SUGGESTIONS_FILE_PATH);
+
+
+        List<Future<Long>> results = ParallelDocumentIndexer.buildAndIndexDocs(
+                new OneDocBuilderImpl(),
+                suggestionsPath,
+                indexService,
+                standaloneServerClient
+        );
+
+        //wait till all indexing done
+        for (Future<Long> each : results) {
+            try {
+                Long genId = each.get();
+                logger.info(String.format("ParallelDocumentIndexer.buildAndIndexDocs returned genId: %s", genId));
+            }
+            catch (ExecutionException | InterruptedException futureException){
+                System.out.println(futureException.getCause());
+            }
+        }
+        long t2 = System.nanoTime();
+
+        System.out.println(
+                String.format("IT took %s nanosecs to index documents", (t2 - t1))
+        );
+
+        // commit
+        standaloneServerClient.getBlockingStub().commit(CommitRequest.newBuilder().setIndexName(INDEX_NAME).build());
+
+    }
+
+    private static BuildSuggestResponse buildSuggester(LuceneServerClient standaloneServerClient){
+        BuildSuggestRequest.Builder buildSuggestRequestBuilder = BuildSuggestRequest.newBuilder();
+        buildSuggestRequestBuilder.setSuggestName("suggest_0");
+        buildSuggestRequestBuilder.setIndexName(INDEX_NAME);
+        buildSuggestRequestBuilder.setInfixSuggester(InfixSuggester.newBuilder().setAnalyzer("default").build());
+        buildSuggestRequestBuilder.setNonLocalSource(SuggestNonLocalSource.newBuilder()
+                .setSuggestField("localized_completed_text")
+                .setWeightField("score")
+                .setContextField("geo_context")
+                .setPayloadField("payload")
+                .build());
+
+        return standaloneServerClient.getBlockingStub().buildSuggest(
+                buildSuggestRequestBuilder.build()
+        );
+    }
+
+    public static void main(String[] args) throws IOException, Exception{
+        // The dir where the indices will live in the remote server
+        Path yelp_suggest_test_base_path = Paths.get("/nail/tmp");
+        // The client who will be talking to the remote server
+        LuceneServerClient standaloneServerClient = new LuceneServerClient("dev83-uswest1adevc", 55886);
```

Comment:
Can we just use `localhost` here instead ?

Reply:
I was planning on making this cmd arg before pushing out this test.